tests/test_peer_funcs.cpp

```diff
@@ -0,0 +1,157 @@
+/*
+    Copyright (c) 2007-2018 Contributors as noted in the AUTHORS file
+
+    This file is part of libzmq, the ZeroMQ core engine in C++.
+
+    libzmq is free software; you can redistribute it and/or modify it under
+    the terms of the GNU Lesser General Public License (LGPL) as published
+    by the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    As a special exception, the Contributors give you permission to link
+    this library with independent modules to produce an executable,
+    regardless of the license terms of these independent modules, and to
+    copy and distribute the resulting executable under terms of your choice,
+    provided that you also meet, for each linked independent module, the
+    terms and conditions of the license of that module. An independent
+    module is a module which is not derived from or based on this library.
+    If you modify this library, you must extend this exception to your
+    version of the library.
+
+    libzmq is distributed in the hope that it will be useful, but WITHOUT
+    ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+    FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public
+    License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "testutil.hpp"
+#include "testutil_unity.hpp"
+
+#include <unity.h>
+
+#ifndef ZMQ_BUILD_DRAFT_API
+#error This test HAS to run only with ZMQ_BUILD_DRAFT_API
+#endif //ZMQ_BUILD_DRAFT_API
+
+void setUp ()
+{
+    setup_test_context ();
+}
+
+void tearDown ()
+{
+    teardown_test_context ();
+}
+
+
+void test_mute_peer ()
```

Comment:
I am not completely sure, but does this contain multiple independent test cases? It would be desirable to split these up, to allow the tests to run and possibly fail independently.

Reply:
No. This is a single test case, as it tests the entire mute functionality (and things are missing; e.g. mute in between a multi-part message or routing-id already read).

Splitting this in multiple subtestcases only bloats the code, as it needs init/finit sequence on each test case, or you end up cascading test cases (without fini/init), in which case the main reason to split - "to allow the tests to run and possibly fail independently" - is not anymore valid.

Reply:
Ok